tools/cli/main.cpp

```diff
@@ -12,246 +12,859 @@
+
+static struct arg_file *xtract_err_f = arg_file0(ERR_SHORT_OPT, ERR_LONG_OPT,
+                                                 FILE_DATATYPE, ERR_GLOSSARY);
+
+static struct arg_str *xtract_output_type = arg_str0(OUT_TYPE_SHORT_OPT,
+                                                     OUT_TYPE_LONG_OPT,
+                                                     TYPE_DATATYPE,
+                                                     OUT_TYPE_GLOSSARY1);
+
+
+static struct arg_file *xtract_catalog_fs = arg_filen(CATALOG_SHORT_OPT,
+                                                      CATALOG_LONG_OPT,
+                                                      FILE_DATATYPE,
+                                                      0,
+                                                      MAX_NO_INPUTS,
+                                                      CATALOG_GLOSSARY1);
+
+static struct arg_str *xtract_symtable_name = arg_str1(SYM_TABLE_NAME_SHORT_OPT,
+                                                       SYM_TABLE_NAME_LONG_OPT,
+                                                       NAME_DATATYPE,
+                                                       SYM_TABLE_NAME_GLOSSARY);
+
+static struct arg_str *xtract_symtable_version = arg_str1(SYM_TABLE_VERSION_SHORT_OPT,
+                                                          SYM_TABLE_VERSION_LONG_OPT,
+                                                          VERSION_DATATYPE,
+                                                          SYM_TABLE_VERSION_GLOSSARY);
+
+static struct arg_file *xtract_input_fs = arg_filen(NULL, NULL, FILE_DATATYPE, 0, MAX_NO_INPUTS, INPUT_GLOSSARY);
+
+static struct arg_end *xtract_end = arg_end(20);
+
+static void *xtract_argtable[] =
+        {xtract, xtract_out_f, xtract_err_f, xtract_output_type, xtract_catalog_fs, xtract_symtable_name,
+         xtract_symtable_version, xtract_input_fs, xtract_end};
+
+static int xtract_nerrors;
+
+static void **all_arg_tables[] = {argtable, proc_argtable, comp_argtable, xtract_argtable, help_argtable,
+                                 version_argtable};
+static int all_arg_tables_size = sizeof(all_arg_tables) / sizeof(all_arg_tables[0]);
+
+//
+// Printing functions
+//
+
+void ion_print_process_examples() {
+    std::string s =
+            //--------------------------------------------------------------------------------
+            //<-                                80 chars                                    ->
+            //--------------------------------------------------------------------------------
+            "Read input.10n and pretty-print it to stdout.\n"
+            "\t $ ion process input.10n\n\n"
+
+            "Read input.ion (using a catalog comprised of the shared symbol tables contained \n"
+            "in catalog.10n) without re-writing, and write a performance report.\n"
+            "\t$ ion process --output-format none --catalog catalog.10n --perf-report report -- input.10n\n\n"
```

Comment:
(This comment applies to all examples) It looks like argtable3 requires a `=` between the option name and value, e.g. `--output-format=none` as opposed to `--output-format none`, is that correct?

If so, we'll need to update the examples and also [ion-test-driver](https://github.com/amzn/ion-test-driver).

Reply:
It works both with as well as without 
```
therapon@u0164d560980059 > echo 'abc' | ./build/release/tools/cli/ion  process --output-format=binary -                                                    [~/Amazon/Work/GitHub/Ion/ion-c]
��遃և��abcq
therapon@u0164d560980059 > echo 'abc' | ./build/release/tools/cli/ion  process --output-format binary -                                                    [~/Amazon/Work/GitHub/Ion/ion-c]
��遃և��abcq

```

Reply:
Excellent!